packages/nodejs/src/config.ts

```diff
@@ -14,17 +15,20 @@ import { ENV_TO_KEY_MAPPING, PRIVATE_ENV_MAPPING } from "./config/configmap"
  */
 export class Configuration {
   data: Partial<AppsignalOptions>
+  sources: HashMap<Partial<AppsignalOptions>>
 
   constructor(options: Partial<AppsignalOptions>) {
-    writePrivateConstants()
-
-    this.data = {
-      ...this._defaultValues(),
-      ...this._loadFromEnvironment(),
-      ...options
+    this.sources = {
```

Comment:
Can you add some tests to test if these values are properly set? In the diagnose tests we will test if the values are set at all, with some variations, but we need to unit test in the integration itself if the sources are set as expected when they are loaded from different sources.

Reply:
Done! Ended up reorganising the config tests quite a bit, removing some redundant tests and adding checks about the sources to existing tests. There's no tests specifically about the sources, but there's tests about how values are overriden, and the resulting state of the sources is checked in those tests.

Reply:
I think we can maybe test the rest as part of https://github.com/appsignal/appsignal-nodejs/issues/489